### ISSUE TITLE
Branch name in the list of repositories

### ISSUE THREAD

**Issue by Redhunt2013** [desktop/desktop#10540]
### Describe the feature or problem you’d like to solve

In the list of repositories view, show the current checked out branch.  We work on many repositories at a time with a developer branch checked out on some of those repo.  Often times, I feel showing the branch name right next to the repo name in the view will let me know whether I have my branch checked out or not.

### Proposed solution

It will benefit the developers working on multiple code repos.

### Additional context

I created a screen mockup.  The idea here is similar to the GitToolBox plugin used in IntelliJ.

![Show Branch Name](https://user-images.githubusercontent.com/5237704/92634839-bd625180-f289-11ea-8ca9-d8aca0293664.png)

**Comment by tierninho** [comment]
@Redhunt2013 Thanks for the suggestion. Note, this is a duplicate request of https://github.com/desktop/desktop/issues/8158

To give you some background, we recently did some work recently to improve the repos list in #6460. We knew that this wouldn't solve every use case and that some folks would be upset with where we ended up. We're not going to do further work on this area of the app in the near future, but we're definitely open to revisiting as we see how others are using the new layout and what's falling short. So I'm going to close this issue to indicate that we don't intend to build anything for this case in the near term, but again we're not opposed to it philosophically so we're open to revisiting down the road. Thanks again!
